[Bug] run job failed: Invalid signature file digest for Manifest main attributes

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### Java Version

1.8

### Scala Version

2.12.x

### StreamPark Version

2.0.0

### Flink Version

Flink1.15.3

### deploy mode

remote

### What happened

run flink job with remote mode, Caused by:  java.lang.SecurityException: Invalid signature file digest for Manifest main attributes

### Error Exception

```log
java.util.concurrent.CompletionException: java.lang.reflect.InvocationTargetException
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1592)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.streampark.flink.client.FlinkClient$.$anonfun$submit$1(FlinkClient.scala:60)
	at org.apache.streampark.flink.proxy.FlinkShimsProxy$.$anonfun$proxy$1(FlinkShimsProxy.scala:63)
	at org.apache.streampark.common.util.ClassLoaderUtils$.runAsClassLoader(ClassLoaderUtils.scala:40)
	at org.apache.streampark.flink.proxy.FlinkShimsProxy$.proxy(FlinkShimsProxy.scala:63)
	at org.apache.streampark.flink.client.FlinkClient$.submit(FlinkClient.scala:55)
	at org.apache.streampark.flink.client.FlinkClient.submit(FlinkClient.scala)
	at org.apache.streampark.console.core.service.impl.ApplicationServiceImpl.lambda$start$10(ApplicationServiceImpl.java:1533)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	... 3 more
Caused by: org.apache.flink.client.program.ProgramInvocationException: The program's entry point class 'org.apache.streampark.flink.cli.SqlClient' caused an exception during initialization: Invalid signature file digest for Manifest main attributes
	at org.apache.flink.client.program.PackagedProgram.loadMainClass(PackagedProgram.java:503)
	at org.apache.flink.client.program.PackagedProgram.<init>(PackagedProgram.java:153)
	at org.apache.flink.client.program.PackagedProgram.<init>(PackagedProgram.java:65)
	at org.apache.flink.client.program.PackagedProgram$Builder.build(PackagedProgram.java:691)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.getJobGraph(FlinkSubmitTrait.scala:198)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.getJobGraph$(FlinkSubmitTrait.scala:189)
	at org.apache.streampark.flink.client.impl.RemoteSubmit$.jobGraphSubmit(RemoteSubmit.scala:120)
	at org.apache.streampark.flink.client.impl.RemoteSubmit$.$anonfun$doSubmit$2(RemoteSubmit.scala:48)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.$anonfun$trySubmit$5(FlinkSubmitTrait.scala:179)
	at scala.util.Try$.apply(Try.scala:209)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.$anonfun$trySubmit$3(FlinkSubmitTrait.scala:179)
	at scala.util.Failure.getOrElse(Try.scala:218)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.trySubmit(FlinkSubmitTrait.scala:177)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.trySubmit$(FlinkSubmitTrait.scala:168)
	at org.apache.streampark.flink.client.impl.RemoteSubmit$.doSubmit(RemoteSubmit.scala:48)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.submit(FlinkSubmitTrait.scala:116)
	at org.apache.streampark.flink.client.trait.FlinkSubmitTrait.submit$(FlinkSubmitTrait.scala:62)
	at org.apache.streampark.flink.client.impl.RemoteSubmit$.submit(RemoteSubmit.scala:38)
	at org.apache.streampark.flink.client.FlinkClientHandler$.submit(FlinkClientHandler.scala:29)
	at org.apache.streampark.flink.client.FlinkClientHandler.submit(FlinkClientHandler.scala)
	... 15 more
Caused by: java.lang.SecurityException: Invalid signature file digest for Manifest main attributes
	at sun.security.util.SignatureFileVerifier.processImpl(SignatureFileVerifier.java:330)
	at sun.security.util.SignatureFileVerifier.process(SignatureFileVerifier.java:263)
	at java.util.jar.JarVerifier.processEntry(JarVerifier.java:318)
	at java.util.jar.JarVerifier.update(JarVerifier.java:230)
	at java.util.jar.JarFile.initializeVerifier(JarFile.java:383)
	at java.util.jar.JarFile.ensureInitialization(JarFile.java:612)
	at java.util.jar.JavaUtilJarAccessImpl.ensureInitialization(JavaUtilJarAccessImpl.java:69)
	at sun.misc.URLClassPath$JarLoader$2.getManifest(URLClassPath.java:991)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:451)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at org.apache.flink.util.ChildFirstClassLoader.loadClassWithoutExceptionHandling(ChildFirstClassLoader.java:71)
	at org.apache.flink.util.FlinkUserCodeClassLoader.loadClass(FlinkUserCodeClassLoader.java:52)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.apache.flink.runtime.execution.librarycache.FlinkUserCodeClassLoaders$SafetyNetWrapperClassLoader.loadClass(FlinkUserCodeClassLoaders.java:172)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.apache.flink.client.program.PackagedProgram.loadMainClass(PackagedProgram.java:479)
	... 34 more
```


### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/streampark-flink/streampark-flink-packer/src/main/scala/org/apache/streampark/flink/packer/maven/MavenTool.scala
+++ b/streampark-flink/streampark-flink-packer/src/main/scala/org/apache/streampark/flink/packer/maven/MavenTool.scala
@@ -20,12 +20,11 @@ package org.apache.streampark.flink.packer.maven
 import java.io.File
 import java.util
 import javax.annotation.{Nonnull, Nullable}
-
 import scala.collection.JavaConversions._
 import scala.collection.mutable.ArrayBuffer
 import scala.util.Try
-
 import com.google.common.collect.Lists
+import org.apache.maven.plugins.shade.filter.Filter
 import org.apache.maven.plugins.shade.{DefaultShader, ShadeRequest}
 import org.apache.maven.plugins.shade.resource.{ManifestResourceTransformer, ResourceTransformer, ServicesResourceTransformer}
 import org.apache.maven.repository.internal.MavenRepositorySystemUtils
@@ -41,7 +40,6 @@ import org.eclipse.aether.spi.connector.transport.TransporterFactory
 import org.eclipse.aether.transport.file.FileTransporterFactory
 import org.eclipse.aether.transport.http.HttpTransporterFactory
 import org.eclipse.aether.util.repository.AuthenticationBuilder
-
 import org.apache.streampark.common.conf.{InternalConfigHolder, Workspace}
 import org.apache.streampark.common.conf.CommonConfig.{MAVEN_AUTH_PASSWORD, MAVEN_AUTH_USER, MAVEN_REMOTE_URL}
 import org.apache.streampark.common.util.{Logger, Utils}
@@ -99,7 +97,6 @@ object MavenTool extends Logger {
       val req = new ShadeRequest
       req.setJars(jarSet)
       req.setUberJar(uberJar)
-      req.setFilters(Lists.newArrayList())
 
       val transformer = ArrayBuffer[ResourceTransformer]()
       // ref https://ci.apache.org/projects/flink/flink-docs-master/docs/connectors/table/overview/#transform-table-connectorformat-resources
@@ -110,6 +107,8 @@ object MavenTool extends Logger {
         transformer += manifest
       }
       req.setResourceTransformers(transformer.toList)
+      // issue: https://github.com/apache/incubator-streampark/issues/2350
+      req.setFilters(List(new ShadeFilter))
       req.setRelocators(Lists.newArrayList())
       req
     }
@@ -208,6 +207,22 @@ object MavenTool extends Logger {
     val repoSystem = newRepoSystem()
     val session = newSession(repoSystem)
     (repoSystem, session)
+  }
+
+  class ShadeFilter extends Filter {
+    override def canFilter(jar: File): Boolean = true
+
+    override def isFiltered(name: String): Boolean = {
+      if (name.startsWith("META-INF/")) {
+        if (name.endsWith(".SF") || name.endsWith(".DSA") || name.endsWith(".RSA")) {
+          logInfo(s"shade ignore file: $name")
+          return true
+        }
+      }
+      false
+    }
+
+    override def finished(): Unit = {}
   }
 
 }

